### PR TITLE
Update Kyber from upstream

### PR DESCRIPTION
**Pull request by thomwiggers**
Makes Kyber-AVX run on MacOS (#251)

#### Manually checked properties

* [x] No stringification macros
* [x] Output-parameter pointers in functions are on the left
* [x] Negative return values on failure of API functions (within restrictions of FO transform).
* [x] variable declarations at the beginning (except in `for (size_t i=...`)
* Optional:
  * [ ] All integer types are of fixed size, using `stdint.h` types (including `uint8_t` instead of `unsigned char`)
  * [ ] Integers used for indexing are of size `size_t`

### CODE REVIEW

**Comment by mouse07410** [comment]
I confirm that with this PR merged, Kyber AVX2 builds successfully on macOS Catalina with the latest released Xcode.

**Comment by thomwiggers** [comment]
Your patch would break Linux support, as the symbols are exported from C without prefix `_` there.

The other test failing is due to #325 landing, and I will get around to rebasing and fixing that.

**Comment by mouse07410** [comment]
OK, **now** it passes the tests. ;-)

Thank you!

**Comment by thomwiggers** [comment]
@mkannwischer although round-3 kyber will soon be a thing, I think we should still merge this as it will probably take a while before we get Kyber-R3 in a mergable shape.

File: crypto_kem/kyber512/avx2/cdecl.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#ifndef PQCLEAN_DILITHIUM2_AVX2_CDECL
```
Comment (jschanck) [inline crypto_kem/kyber512/avx2/cdecl.inc:1]: Bad namespacing here and in other cdecl.inc files.

**Comment by thomwiggers** [comment]
Replaced by #349

**Pull request closed without merging**